### PR TITLE
Made MockDataManager and allowed the project to use it for testing. A…

### PR DESCRIPTION
**Pull request by jstarry95**
…lso improved tests

### CODE REVIEW

File: BoardBank/BoardBankTests/AddPlayerViewModelTests.swift [hunk 1]
```diff
@@ -154,101 +240,11 @@ final class AddPlayerViewModelTests: XCTestCase {
...
   func testIsDuplicate() {
-    //should still exist from addPlayer tests
-    XCTAssertTrue(sut.isDuplicatePlayer(name: "Sir Wigglebottoms"))
+    sut.addPlayer(nameText: "Big Mike and the Gang", balanceText: "350")
+    XCTAssertTrue(sut.isDuplicatePlayer(name: "Big Mike and the Gang"))
     
+    sut.addPlayer(nameText: "Player", balanceText: "2001")
     //should be duplicate because a Player was created above with the name 'Player'. The isDuplicatePlayer function should handle this as empty strings are defaulted to 'Player'
     XCTAssertTrue(sut.isDuplicatePlayer(name: ""))
   }
```
Comment (jstarry95) [inline BoardBank/BoardBankTests/AddPlayerViewModelTests.swift]: Blank line at eof

File: BoardBank/MonoBank/BankManager.swift [hunk 2]
```diff
@@ -60,47 +60,53 @@ final class BankManager {
   
   /// All players of the current game
   var players: [Player] {
-    return coreDataManager.getAllObjects(ofType: Player.self).sorted()
+    return dataManager.getAllObjects(ofType: Player.self).sorted()
   }
   
-  init(userDefaults: UserDefaults = UserDefaults.standard,
-       coreDataManager: CoreDataManager = CoreDataManager.shared) {
+  //basically only for testing
```
Comment (jstarry95) [inline BoardBank/MonoBank/BankManager.swift]: This comment is not needed. Furthermore, it is untrue

File: BoardBank/MonoBank/BankManager.swift [hunk 3]
```diff
@@ -60,47 +60,53 @@ final class BankManager {
...
     //Setup data persistence
     self.userDefaults = userDefaults
-    self.coreDataManager = coreDataManager
+    
+    switch dataManagerType {
+    case .CoreDataManager:
+      self.dataManager = CoreDataManager.shared
+    case .MockDataManager:
+      self.dataManager = MockDataManager.shared
+
```
Comment (jstarry95) [inline BoardBank/MonoBank/BankManager.swift]: No blank line here. Probably removed "default" and forgot

File: BoardBank/MonoBank/CoreDataManager.swift [hunk 4]
```diff
@@ -9,7 +9,7 @@
 import Foundation
 import CoreData
 
-final class CoreDataManager {
+class CoreDataManager: DataManager {
```
Comment (jstarry95) [inline BoardBank/MonoBank/CoreDataManager.swift]: Should still be final

File: BoardBank/MonoBank/DataManagerProtocol.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+protocol DataManager {
+  func getAllObjects<T: NSManagedObject>(ofType: T.Type) -> [T]
+  
+  func createObject<T: NSManagedObject>(ofType: T.Type, shouldInsert: Bool) throws -> T
+  
+  func save() throws
+  
+  func delete(object: NSManagedObject)
+}
```
Comment (jstarry95) [inline BoardBank/MonoBank/DataManagerProtocol.swift]: Blank line at eof

File: BoardBank/MonoBank/DataManagerType.swift [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+//  Created by Consultant on 7/31/19.
+//  Copyright © 2019 Richard Neitzke. All rights reserved.
+//
+
+import Foundation
+
+enum DataManagerType {
+  case CoreDataManager
+  case MockDataManager
+}
```
Comment (jstarry95) [inline BoardBank/MonoBank/DataManagerType.swift]: Blank line at eof

File: BoardBank/MonoBank/MockDataManager.swift [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  
+  func save() throws {
+    
+  }
+  
+  func delete(object: NSManagedObject) {
+    guard let objectIndex = objectsArray.firstIndex(of: object) else { return }
+    objectsArray.remove(at: objectIndex)
+  }
+}
```
Comment (jstarry95) [inline BoardBank/MonoBank/MockDataManager.swift]: Blank line at eof

File: BoardBank/MonoBank/BankManager.swift [hunk 8]
```diff
@@ -60,47 +60,53 @@ final class BankManager {
...
-  init(userDefaults: UserDefaults = UserDefaults.standard,
-       coreDataManager: CoreDataManager = CoreDataManager.shared) {
+  init(userDefaults: UserDefaults = UserDefaults.standard, dataManagerType: DataManagerType = .CoreDataManager) {
     //Setup data persistence
     self.userDefaults = userDefaults
-    self.coreDataManager = coreDataManager
+    
+    switch dataManagerType {
+    case .CoreDataManager:
+      self.dataManager = CoreDataManager.shared
```
Comment (prrenfrew) [inline BoardBank/MonoBank/BankManager.swift]: Remember to only use self when required. It was used before to disambiguate coreDataManager.

File: BoardBank/MonoBank/MockDataManager.swift [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+//  BoardBank
+//
+//  Created by Consultant on 7/31/19.
+//  Copyright © 2019 Richard Neitzke. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+final class MockDataManager: DataManager {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MockDataManager.swift]: Mock looks pretty good. Good job!

File: BoardBank/MonoBank/BankManager.swift (hunk 8 above)
Comment (jstarry95) [reply BoardBank/MonoBank/BankManager.swift]: Yes, sorry I added these when I was frustrated haha. I will remove the self

File: BoardBank/MonoBank/MockDataManager.swift (hunk 9 above)
Comment (jstarry95) [reply BoardBank/MonoBank/MockDataManager.swift]: Thank you, I do my best

**Comment by prrenfrew** [comment]
@ChrisThirtle since there seems to be some overlap between this PR and yours, please coordinate to figure out what you guys want to do.

**Comment by prrenfrew** [comment]
Closing pending resolution of overlapping implementations between @jstarry95  and @ChrisThirtle

**Pull request closed without merging**